lib/readline.js

```diff
@@ -794,16 +849,20 @@ Interface.prototype._moveCursor = function(dx) {
   }
 };
 
+function breakOutOfReverseSearch() {
+  this.inReverseSearch = false;
+  this._refreshLine();
+}
 
 // handle a write from the tty
 Interface.prototype._ttyWrite = function(s, key) {
   const previousKey = this._previousKey;
   key = key || {};
   this._previousKey = key;
 
-  // Ignore escape key, fixes
-  // https://github.com/nodejs/node-v0.x-archive/issues/2876.
-  if (key.name === 'escape') return;
+  if (key.name === 'escape') {
+    breakOutOfReverseSearch.call(this);
```

Comment:
Should this not only be called if `this.inReverseSearch === true`? I would guess it should actually look like:

```js
if (key.name === 'escape') {
  if (this.inReverseSearch) {
    breakOutOfReverseSearch.call(this);
  }
  // Else, ignore escape key. Fixes:
  // https://github.com/nodejs/node-v0.x-archive/issues/2876.
  return;
}
```